tests/ui/job-view/fuzzy_test.jsx

```diff
@@ -0,0 +1,76 @@
+        currentRepo={currentRepo}
+        notify={() => {}}
+      />
+    </Provider>
+  );
+  const initialList =
+    'addon-tps-xpiartifact-build-linux64-artifact/optbuild-android-aarch64-gcp/debugbuild-android-aarch64-gcp/optbuild-android-aarch64/debugbuild-android-aarch64/optbuild-android-aarch64/pgobuild-android-api-16-gcp/debugbuild-android-api-16-gcp/optbuild-android-api-16/debugbuild-android-api-16/optbuild-android-api-16/pgobuild-android-geckoview-docs/optbuild-android-x86-fuzzing/debugbuild-android-x86-gcp/optbuild-android-x86/optbuild-android-x86_64-asan-fuzzing/optbuild-android-x86_64-gcp/debugbuild-android-x86_64-gcp/optbuild-android-x86_64/debugbuild-android-x86_64/optbuild-fat-aar-android-geckoview-fat-aar/optbuild-linux-devedition/optbuild-linux-gcp/debugbuild-linux-gcp/optbuild-linux-reproduced/optbuild-linux-rusttests/debugbuild-linux-rusttests/optbuild-linux-shippable/optbuild-linux/debugbuild-linux/optbuild-linux64-aarch64/optbuild-linux64-add-on-devel/optbuild-linux64-asan-fuzzing-ccov/optbuild-linux64-asan-fuzzing/optbuild-linux64-asan/debugbuild-linux64-asan/optbuild-linux64-ccov/optbuild-linux64-devedition/optbuild-linux64-fuzzing-ccov/optbuild-linux64-fuzzing/debugbuild-linux64-gcp/debugbuild-linux64-gcp/optbuild-linux64-noopt/debugbuild-linux64-plain/debugbuild-linux64-plain/optbuild-linux64-rusttests/debugtest-windows10-64/opt-reftest-fis-e10s-2test-windows10-64/opt-reftest-no-accel-e10s-1test-windows10-64/opt-reftest-no-accel-e10s-2test-windows10-64/opt-reftest-no-accel-e10s-3test-windows10-64/opt-reftest-no-accel-e10s-4test-windows10-64/opt-talos-bcv-e10stest-windows10-64/opt-talos-chrome-e10stest-windows10-64/opt-talos-damp-e10stest-windows10-64/opt-talos-dromaeojs-e10stest-windows10-64/opt-talos-g1-e10stest-windows10-64/opt-talos-g4-e10stest-windows10-64/opt-talos-g5-e10stest-windows10-64/opt-talos-other-e10s';
+  const searchLinuxResults =
+    'build-linux-devedition/optbuild-linux-gcp/debugbuild-linux-gcp/optbuild-linux-reproduced/optbuild-linux-rusttests/debugbuild-linux-rusttests/optbuild-linux-shippable/optbuild-linux/debugbuild-linux/optbuild-linux64-aarch64/optbuild-linux64-add-on-devel/optbuild-linux64-asan-fuzzing-ccov/optbuild-linux64-asan-fuzzing/optbuild-linux64-asan/debugbuild-linux64-asan/optbuild-linux64-ccov/optbuild-linux64-devedition/optbuild-linux64-fuzzing-ccov/optbuild-linux64-fuzzing/debugbuild-linux64-gcp/debugbuild-linux64-gcp/optbuild-linux64-noopt/debugbuild-linux64-plain/debugbuild-linux64-plain/optbuild-linux64-rusttests/debugartifact-build-linux64-artifact/opt';
+
+  test('Fuzzy search gives expected results', async () => {
+    const { getByTitle, getByTestId } = await render(testFuzzyJobFinder);
+    const inputElement = getByTitle('Filter the list of runnable jobs');
+    const fuzzySearchList = getByTestId('fuzzyList');
+
+    expect(fuzzyJobList).toHaveLength(60);
+    await waitFor(() => expect(fuzzySearchList).toHaveTextContent(initialList));
```

Comment:
Nit: Since we're comparing a list of results`initialResults` and the other strings you have for the expected results should be in list (array) format. Then you can use the jest matcher `arrayContaining` (you're not limited to using the custom jest matchers - such as `toHaveTextContent` for use with the react-testing-library. You can use all jest matchers).

Reply:
@SuyashSalampuria Can you address this nit and let me know when you have (I don't get alerts when you push new commits so you either need to request a review again or mention me).

Reply:
I was just looking over it yesterday
I was a bit confused as how should I compare a DOM to be an array 
We expect the value of this
```<Input
                type="select"
                multiple
                onChange={this.updateAddButton}
                data-testid="fuzzyList"
              >
                {this.state.fuzzyList.sort(sortAlphaNum).map((e) => (
                  <option
                    title={`${e.name} - ${e.groupsymbol}(${e.symbol})`}
                    key={e.name}
                    className={
                      this.state.selectedList.includes(e.name) ? 'selected' : ''
                    }
                  >
                    {e.name}
                  </option>
                ))}
              </Input>
```
How can I compare the content of this DOM to an array 
Could you help me in this @sarah-clements

Reply:
What I meant was that you can select all of the `option` nodes instead of the fuzzy list element so you can: 
1) assert on the length of that and compare it to `fuzzyJobList` (right now you're asserting the length on the `fuzzyJobList` but it's not clear to me why)

2)  iterate over it the array of option nodes via map to extract the title into an array of strings. Although admittedly that might be a bit more code. 

If your approach turns out to be simpler it's fine to keep it but at the very least you should programmatically convert `initialList` into a string from the json file rather than hard-code it.

Reply:
Yeah, it should be pretty easy to do `fuzzyJobList.options.map( x => x.textContent)` to get an array of the option names to compare against in the tests.